scraper_cleveland.py

```diff
@@ -0,0 +1,310 @@
+    JOB_VALUE_XPATH = '//div[span="Job Value($):"]/span[2]'
+    NATURE_OF_JOB_XPATH = '//div[span="Nature of Job Desc: "]/following-sibling::div'
+    PROJECT_DESCRIPTION = '//div[h1="Project Description:"]/span'
+    DESCRIPTIVE_NATURE_XPATH = '//div[span="Descriptive nature of complaint: "]/following-sibling::div'
+
+
+DETAILS_MAP = {  # the constant maps the attribute needed to scrape to the referring content xpath
+    DetailedHeader.STREET_ADDRESS.value: [DetailedElement.WORK_LOCATION_XPATH.value],
+    DetailedHeader.CONSIDERATION.value: [DetailedElement.JOB_VALUE_XPATH.value],
+    DetailedHeader.GRANTEE.value: [DetailedElement.OWNER_XPATH.value],
+    DetailedHeader.DESCRIPTION.value: [
+        DetailedElement.NATURE_OF_JOB_XPATH.value,
+        DetailedElement.PROJECT_DESCRIPTION.value,
+        DetailedElement.DESCRIPTIVE_NATURE_XPATH.value
+    ]
+}
+
+
+def basic_clean(text: str) -> str:
+    """
+    Function do a basic clean of the text scraped off the web site, including strip the text off whitespaces,
+    linebreaks, ect.
+    :param text: input the text to be cleaned
+    :return: return the cleaned text
+    """
+    return text.strip().replace('\r\n', ', ').replace('\r', ', ').replace('\n', ', ')
+
+
+def wait_for_staleness(driver, element_id, timeout=120):
+    """
+    The function waits until the old web element detaches the DOM after loading a new page. Otherwise, the script won't
+    get the new element and the function raises TimeOut Exception.
+    :param driver: the web driver in use
+    :param element_id: the id of the web element to be inspected
+    :param timeout: wait a timeout period of time for the web element to detach the DOM, defaulted to 10 seconds.
+    :return: raise exceptions and exit or no returns
+    """
+    try:
+        WebDriverWait(driver, timeout).until(
+            EC.staleness_of(driver.find_element_by_id(element_id))
+        )
+    except Exception as e:
+        print(e)
+        exit(1)
+
+
+def wait_for_element_load(driver: webdriver, by_method: By, method_val: str, timeout=120) -> webdriver:
+    """
+    The function waits until the expected web element presented in the page to proceed to next
+    :param driver: the web driver in use
+    :param by_method: a `By` object used to locate the desired web element in by different method
+    :param method_val: the values of the `By` method
+    :param timeout: wait a timeout period of time for the web element to be presented in the page,
+                    defaulted to 10 seconds.
+    :return: raise exceptions or return the element be waited on
+    """
+    try:
+        element = WebDriverWait(driver, timeout).until(
+           EC.presence_of_element_located((by_method, method_val))
+        )
+    except Exception as e:
+        print(e)
+        exit(1)
+    else:
+        return element
+
+
+def wait_for_elements_load(driver: webdriver, by_method: By, method_val: str, timeout=120) -> list:
```

Comment:
Comment should be different than the comment for `wait_for_element_load `

Reply:
fixed